### ISSUE TITLE
CPU usage is too high during the training phas, and the GPU usage is too low

### ISSUE THREAD

**Issue by SuHe36** [grammarly/gector#66]
When I train the model at the stageI and stageII, I find that the CPU usage is too high(2700%), and the GPU usage is too low(15%).
The machine is V100.

Does anybody know what causes this phenomenon ?
thanks for any reply .

**Comment by SuHe36** [comment]
I found the number of threads is more than 30. What should I do to change the number of threads ?

**Comment by skurzhanskyi** [comment]
I believe the CPU usage is high because of the BPE/WordPiece tokenization 🤔 
It can be related to the AllenNLP library as well.

**Comment by aonurK** [comment]
Hi!
I recognized the same issue, and I did not dig into the codes for now. If I dont get it wrong, the GPU usage by python is almost zero while CPU is 100%. I debugged a little bir and saw that cuda device is available as well.  I'm not sure whether GPU is utilized or not.Is this what we expect or am I using the CPU only.     

In the beginning of the training, it/s value starts with ~30, but it decreased slowly and now 41% of one epoch is finished and this number became ~10.
